pkg/v3/types/basetypes.go

```diff
@@ -1,10 +1,32 @@
 package types
 
 import (
+	"encoding/hex"
 	"fmt"
 	"math/big"
+	"strings"
 )
 
+var checkResultStringTemplate = `{
+	"PipelineExecutionState":%d,
+	"Retryable":%v,
+	"Eligible":%v,
+	"IneligibilityReason":%d,
+	"UpkeepID":%s,
+	"Trigger":%s,
+	"WorkID":"%s",
+	"GasAllocated":%d,
+	"PerformData":"%s",
+	"FastGasWei":%s,
+	"LinkNative":%s
+}`
+
+func init() {
+	checkResultStringTemplate = strings.Replace(checkResultStringTemplate, " ", "", -1)
+	checkResultStringTemplate = strings.Replace(checkResultStringTemplate, "\t", "", -1)
+	checkResultStringTemplate = strings.Replace(checkResultStringTemplate, "\n", "", -1)
+}
```

Comment:
not sure if this is a common pattern we don't use it in other packages. I don't have much experience here though will defer to @ferglor or @amirylm

Reply:
oh wow you're right. I think this is a pretty common go-ism, though I can remove it if y'all prefer. fwiw it seems we use init functions all over core.

Reply:
Yeah I think using init is fine; I take it the idea here is to ensure the templated string is collapsed into a single line for printing?

Reply:
@ferglor yeah that was the idea